### ISSUE TITLE
"Error: sensor frame is corrupted is uDriver powered on?" even if the uDriver is powered on

### ISSUE THREAD

**Issue by emrecemal** [thomasfla/odri-spi-rpi#1]
Hello,

I'm trying to run this example, but right now I'm getting this error even if the uDriver is on. What can be the problem?

![IMG_9027](https://github.com/user-attachments/assets/fa2ba052-bd36-4d12-a1b8-c536e9f6f21d)

**Comment by thomasfla** [comment]
Did you flash the udrivers with spi firmware (not CAN)?
did you enabled SPI on raspberry pi using the raspi-config ? https://www.raspberrypi-spy.co.uk/2014/08/enabling-the-spi-interface-on-the-raspberry-pi/
You can debug by printing the response SPI packet in the python code

**Comment by emrecemal** [comment]
Yes, I did all. I discovered that the issue was with the cabling. I resolved the problem and successfully ran the test code. However, during the calibration process, the code asked me to manually rotate the motors. It was able to detect Index 1, but not Index 0. Since the actuator modules were recently assembled, I suspect that the issue may be related to either the encoder or the polarity of the brushless motor. I set the flag ```absolutePositionMode``` to ```False```, and kept going. The motor was not smooth. I'm currently investigating these potential causes. Any thoughts or suggestions on how to proceed?

**Comment by thomasfla** [comment]
The code in this repo is for controlling two motors, it is not very flexible and used for teaching with a dual motor test bed: 
https://github.com/open-dynamic-robot-initiative/open_robot_actuator_hardware/blob/master/mechanics/dual_motor_testbed_v1/README.md#dual-motor-testbed-v1
It is however a good proof of concept and valid starting point.

Here are a few suggestions according to your problems:

If you are using only one motor, you need to adapt the code and skip calibration of the other one.

If you are using actuator module (so with the gearbox), you will have to tune the gains Kp and Kd.

If the motor does not turn at all and just shake, you will have to swap two phases.

I hope this helps !
